### PR TITLE
Support input NV12 textures

### LINKED ISSUE #750: Add NV12 format conversion to prepare the compositor for ingesting output from the vulkan decoder

### PR DESCRIPTION
**Pull request by jerzywilczek**
Closes #750

### CODE REVIEW

File: compositor_render/src/wgpu/texture.rs [hunk 1]
```diff
@@ -182,6 +189,15 @@ impl InputTexture {
                             error!("Invalid texture passed as an input: {err}")
                         }
                     }
+                    InputTextureState::Nv12WgpuTexture(texture) => {
+                        let texture = NV12TextureView::from_wgpu_texture(texture.as_ref());
+                        let bind_group = texture.new_bind_group(ctx, ctx.format.nv12_layout());
```
Comment (wkozyra95) [inline compositor_render/src/wgpu/texture.rs:194]: what will happen if the passed texture is not nv12?

e.g. for rgba texture with do some validation inside fill_from_wgpu_texture

**Review by wkozyra95 — APPROVED** [review]
LGTM, but make sure it does not panic when passed texture is invalid

File: compositor_render/src/wgpu/texture/nv12.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    binding: 0,
+                    resource: wgpu::BindingResource::TextureView(&self.view_y),
+                },
+                wgpu::BindGroupEntry {
+                    binding: 1,
+                    resource: wgpu::BindingResource::TextureView(&self.view_uv),
+                },
+            ],
+        })
+    }
```
Comment (WojciechBarczynski) [inline compositor_render/src/wgpu/texture/nv12.rs:82]: This can be created just once

**Review by WojciechBarczynski — APPROVED** [review]

File: compositor_render/src/wgpu/texture/nv12.rs (hunk 2 above)
Comment (jerzywilczek) [reply compositor_render/src/wgpu/texture/nv12.rs:82]: and it is indeed created just once. all texture types have a similar function and all of them are created just once

File: compositor_render/src/wgpu/texture.rs (hunk 1 above)
Comment (jerzywilczek) [reply compositor_render/src/wgpu/texture.rs:200]: ok, handled that

**Pull request merged**